components/server/src/workspace/headless-log-controller.ts

```diff
@@ -412,6 +343,52 @@ export class HeadlessLogController {
     }
 }
 
+function createStreamingResponseWriter(logCtx: LogContext, res: express.Response, taskId: string) {
```

Comment:
This is a great abstraction, long overdue! The only nit I have here is calling the parameter `taskId` when we actually send a terminal ID in the headless logs case.

Reply:
Jup, found that to be an ok tradeoff, as that endpoint shouldn't be used anymore.